### PR TITLE
bencodetools: fix build

### PR DESCRIPTION
**Pull request by ocfox**
## Things done
Fix https://hydra.nixos.org/build/265499967

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/bencodetools/default.nix [hunk 1]
```diff
@@ -21,12 +22,14 @@ stdenv.mkDerivation rec {
       --replace 'python_install_option=""' 'python_install_option="--prefix=$out"'
   '';
 
-  enableParallelBuilding = true;
```
Comment (OPNA2608) [inline pkgs/development/libraries/bencodetools/default.nix:24]: This is not default-on in the base `stdenv.mkDerivation`. If you need to turn this off for some reason, please make it explicitly `enableParallelBuilding = false` and add a comment with an explanation. As is, this gets rid of parallelism:

```
build flags: -j2 SHELL=/nix/store/m101dg80ngyjdb02g6jwy80sr7kzj26g-bash-5.2p26/bin/bash
```
->
```
build flags: SHELL=/nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/bash
```

File: pkgs/development/libraries/bencodetools/default.nix [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-, python3
+{
+  stdenv,
+  lib,
+  fetchFromGitLab,
+  python3,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation {
```
Comment (OPNA2608) [inline pkgs/development/libraries/bencodetools/default.nix:8]: Could you put these kinds of style updates into a separate commit please?

File: pkgs/development/libraries/bencodetools/default.nix (hunk 1 above)
Comment (ocfox) [reply pkgs/development/libraries/bencodetools/default.nix:24]: I mistakenly remembered this as the default option.

**Comment by OPNA2608** [comment]
```
       … while evaluating attribute 'nativeBuildInputs' of derivation 'bencodetools-unstable-2022-05-11'

       … while evaluating derivation 'python3-3.12.5-env'

       … while calling 'requiredPythonModules'

       … while evaluating attribute 'nativeBuildInputs' of derivation 'python3.12-distutils-72.1.0'

       error: Package ‘python3.12-jaraco-path-3.7.1’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/macstadium-x86-44911362/pkgs/development/python-modules/jaraco-path/default.nix:30 is marked as broken, refusing to evaluate.
```

Welp, RIP Darwin >\~>".

**Review by OPNA2608 — APPROVED** [review]
Darwin falling away is unfortunate, but prolly out-of-scope for this PR.

**Pull request merged**